### PR TITLE
feat: Show unauthorized state feature

### PR DESCRIPTION
**Pull request by adi6859**
# Description
For non-superadmin users show unauthorized state.
Same is the case with
- git-account
- container-registry
- chart repository
- host URL
- projects
Basically in this I have to show unauthorized state for non-superadmin by showing error 403 
Fixes # [(AB1934)](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/CSAT/Stories/?workitem=1934)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

All the test are done manually .
1.By changing the condition for superadmin.
2.I have used isSuperAdmin variable for all checks and validation.
3. isSuperAdmin is a boolean variable.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/hostURL/HostURL.tsx [hunk 1]
```diff
@@ -138,6 +138,14 @@ export default class HostURLConfiguration extends Component<HostURLConfigProps,
         if (this.state.view === ViewType.LOADING) {
             return <Progressing pageLoader />
         }
+        else if(!this.props.isSuperAdmin){
```
Comment (milindk8) [inline src/components/hostURL/HostURL.tsx:141]: Club both error condition of ErrorScreenManager into one

File: src/components/chartRepo/ChartRepo.tsx [hunk 2]
```diff
@@ -13,52 +13,137 @@ import "./chartRepo.scss";
...
     const [loading, result, error, reload] = useAsync(getChartRepoList)
-    const [fetching, setFetching] = useState(false);
+    const [fetching, setFetching] = useState(false)
     if (loading && !result) return <Progressing pageLoader />
     if (error) {
         showError(error)
         if (!result) return null
     }
+    
+    function getRandomInt(): number {
```
Comment (milindk8) [inline src/components/chartRepo/ChartRepo.tsx:25]: can't we reduce the function code ?

File: src/components/chartRepo/ChartRepo.tsx [hunk 3]
```diff
@@ -13,52 +13,137 @@ import "./chartRepo.scss";
 import DeleteComponent from '../../util/DeleteComponent';
 import { DC_CHART_REPO_CONFIRMATION_MESSAGE, DeleteComponentsName } from '../../config/constantMessaging';
 
-export default function ChartRepo() {
+export default function ChartRepo({ isSuperAdmin }) {
```
Comment (shivani170) [inline src/components/chartRepo/ChartRepo.tsx:16]: type is missing

File: src/components/chartRepo/ChartRepo.tsx [hunk 4]
```diff
@@ -13,52 +13,137 @@ import "./chartRepo.scss";
...
+                <p className="form__subtitle">
+                    Manage your organization’s chart repositories.
+                    <span>
+                        <a
+                            rel="noreferrer noopener"
+                            target="_blank"
+                            className="dc__link"
+                            href={DOCUMENTATION.GLOBAL_CONFIG_CHART}
+                        >
+                            {' '}
```
Comment (shivani170) [inline src/components/chartRepo/ChartRepo.tsx:70]: Remove all {' '}

File: src/components/chartRepo/ChartRepo.tsx [hunk 5]
```diff
@@ -13,52 +13,137 @@ import "./chartRepo.scss";
...
+                    <div className="chartRepo_form__subtitle dc__float-right">
+                        <a
+                            rel="noreferrer noopener"
+                            target="_blank"
+                            className={`dc__link ${fetching ? '' : 'cursor'}`}
+                            onClick={refetchCharts}
+                        >
+                            <span>
+                                <svg
+                                    width="16"
```
Comment (shivani170) [inline src/components/chartRepo/ChartRepo.tsx:98]: import svg instead

**Review by shivani170 — APPROVED** [review]

**Review by milindk8 — APPROVED** [review]

File: src/components/chartRepo/ChartRepo.tsx (hunk 2 above)
Comment (adi6859) [reply src/components/chartRepo/ChartRepo.tsx:30]: not sure

**Review by shivani170 — APPROVED** [review]

File: src/config/constantMessaging.ts [hunk 6]
```diff
@@ -80,6 +80,8 @@ export const ERROR_EMPTY_SCREEN = {
     ALL_SET_GO_CONFIGURE: 'Looks like you’re all set. Go ahead and select an image to deploy',
     DEPLOYEMENT_WILL_BE_HERE: 'Once deployed, details for the deployment will be available here',
     GO_TO_DEPLOY: 'Go to deploy',
+    ONLY_FOR_SUPERADMIN:'Information on this page is available only to superadmin users.',
+    NOT_AUTHORIZED:"Not authorized",
```
Comment (milindk8) [inline src/config/constantMessaging.ts:84]: format this file

**Review by milindk8 — APPROVED** [review]

**Pull request merged**